manager.go

```diff
@@ -169,6 +133,19 @@ func (manager *resolveManager) resolve(req resolveRequest) {
 	req.response <- resolverResponse{name: req.name, result: result, err: err}
 }
 
-func (manager *resolveManager) stop() {
-	manager.once.Do(func() { close(manager.quit) })
+func (manager *resolveManager) resolveRequest(name string, response chan<- resolverResponse, fn FieldResolveFn, params ResolveParams) {
+	req := resolveRequest{
+		fn:       fn,
+		name:     name,
+		params:   params,
+		response: response,
+	}
+
+	select {
+	case manager.resolveRequests <- req:
+		return
+	default:
+		go manager.newWorker()
+		manager.resolveRequests <- req
```

Comment:
Couldn't there be a race condition with this? Like - before you check to see if you can send in more to the buffer, but here you don't. Hypothetically you could have another request come in and take the `newWorker` which is a small case, but then we have to wait until one of the `infiniteWorker`s can get to it. Otherwise it can cause deadlock.

Reply:
It is a buffered channel so if it can't add a new request onto the channel then all of the existing workers are busy and the buffer is full. That means that when the new worker starts up it will take a message off the channel which was one which was already in the buffer. Then either that will free up space in the buffer for this message or another message will sneak in first. Either way this is in line and there is one more worker processing and none of the workers exit until the buffer is empty.

Reply:
Sure. it just feels... a little weird. I dunno.